internal/p2p/rqueue_test.go

```diff
@@ -0,0 +1,47 @@
+package p2p
+
+import (
+	"context"
+	"testing"
+	"time"
+)
+
+func TestSimpleQueue(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// set up a small queue with very small buffers so we can
+	// watch it shed load, then send a bunch of messages to the
+	// queue, most of which we'll watch it drop.
+	sq := newSimplePriorityQueue(ctx, 1, nil)
+	for i := 0; i < 100; i++ {
+		sq.enqueue() <- Envelope{From: "merlin"}
```

Comment:
I believed that the priority ordering would be guaranteed by the priority heap which I think is tested by the heap itself, which is tested elsewhere. 

The feature I was going after was more "will this shed load without blocking" and less "is the priority ordering correct," (which I think may never matter, because either the network is contended or its not.)

Reply:
(I think this follows from the thread above)